sort/mergeSort.go

```diff
@@ -2,204 +2,157 @@ package sort
 
 import (
 	"container/heap"
-	"fmt"
-	"github.com/vertgenlab/gonomics/axt"
-	"github.com/vertgenlab/gonomics/bed"
-	"github.com/vertgenlab/gonomics/common"
+	"encoding/gob"
 	"github.com/vertgenlab/gonomics/exception"
-	"github.com/vertgenlab/gonomics/fileio"
-	"github.com/vertgenlab/gonomics/giraf"
-	"github.com/vertgenlab/gonomics/sam"
-	"github.com/vertgenlab/gonomics/vcf"
-	"log"
+	"io"
 	"os"
-	"path"
 	"sort"
 )
 
-const (
-	// Newly added filetypes should be added to the following valid types list
-	validFileTypes     = "axt, bed, vcf, sam, giraf"
-	maxTmpFilesAllowed = 1000
-)
+type heapBuf []any
```

Comment:
You save so many lines of code

Reply:
I'm learning here that the generic type can just be declared with this 1 line. I thought you had to specify it somehow, but it seems like that's just because the example in A Tour of Go was a "singly-linked list" that requires a pointer and a value